pkg/monitors/haproxy/helpers.go

```diff
@@ -262,10 +261,12 @@ func parseStatusField(v string) int64 {
 }
 
 func httpReader(conf *Config, method string) (io.ReadCloser, error) {
-	client := http.Client{
-		Timeout:   conf.Timeout,
-		Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: !conf.SSLVerify}},
+	client, err := conf.Build()
+	if err != nil {
+		return nil, err
 	}
+	defer client.CloseIdleConnections()
```

Comment:
This should really be refactored to reuse http.Client. I started but haven't finished. It's no worse than it was before though.